### ISSUE TITLE
Text rendering artifact

### ISSUE THREAD

**Issue by coffeemug** [vadimdemedes/ink#529]
The following component:

```jsx
<Box flexDirection="column" width={80}>
  <Box>
    <Text>📈🚀</Text>
    <Text color="green">test1</Text>
    <Text color="gray">test2</Text>
    <Text>test3</Text>
  </Box>
  <Box>
    <Text>test4</Text>
  </Box>
</Box>
```

Produces the following image:

<img width="171" alt="image" src="https://user-images.githubusercontent.com/48436/182964498-9c20aaab-cd3d-473c-b199-5156eb85902d.png">

The artifact here is that `test4` has a strikethrough style applied to it. It shouldn't.

This appears to be a minimum example; every line of that component seems necessary. If I remove the line with `test1` or `test2` or `test3`, everything renders fine.

Note that the strikethrough artifact appears only after the box that contains the emoji ends (all text after that is stroke through). But somehow the text in the same box as the emoji isn't.

**Comment by vadimdemedes** [comment]
Traced it down to [slice-ansi](https://github.com/chalk/slice-ansi):

```js
import sliceAnsi from 'slice-ansi';

// "📈🚀XY" where "X" is green and "Y" is gray
const str = '📈🚀\x1B[32mX\x1B[39m\x1B[90mY\x1B[39m';
sliceAnsi(str, 0, 4)
//=> 📈🚀\x1B[32mX\x1B[39m\x1B[90mY\x1B[29m
```

Note the `[29m` at the end instead of `[39m` in the original `str` variable. I'm not an expert in ANSI codes, but I suspect it has something to do with it.

**Comment by vadimdemedes** [comment]
Fixed via #565.
